src/react/useMutation.ts

```diff
@@ -97,9 +93,9 @@ export function useMutation<
   }, [])
 
   const mutate = React.useCallback<
-    UseMutateFunction<TData, TError, TVariables, TContext>
+    MutateFunction<TData, TError, TVariables, TContext>
   >((variables, mutateOptions) => {
-    obsRef.current!.mutate(variables, mutateOptions).catch(noop)
+    return obsRef.current!.mutate(variables, mutateOptions).catch(noop)
   }, [])
```

Comment:
why this change please? It's per design that `mutate` doesn't return anything...

Reply:
MutationFunction (core/types) and UseMutateFunction (react/types) are quite identical. The only difference is that MutationFuncation returns Promise<TData>, and useMutationFunction returns void. So I assumed that MutationFunction is more accurate since it's from the core package. 

If this isn't true, I'm happy to revert the change

Reply:
yes please. `.mutate` shouldn't return a promise. we've had that in v2 and it was confusing for many, which is why we added a dedicated `mutateAsync` for that case.

Reply:
what's the status here please? There seem to be conflicts now with the new release we just made, please have a look at resolving them :)

Reply:
sorry to keep this on hold, Dominik

I have reverted the .mutate change and solved the conflict, could you please have a look now?

thanks for your time.